core/src/PrognosticData.cpp

```diff
@@ -22,10 +22,10 @@ PrognosticData::PrognosticData()
     , pOcnBdy(0)
 
 {
-    registerProtectedArray(ProtectedArray::H_ICE, &m_thick);
-    registerProtectedArray(ProtectedArray::C_ICE, &m_conc);
-    registerProtectedArray(ProtectedArray::H_SNOW, &m_snow);
-    registerProtectedArray(ProtectedArray::T_ICE, &m_tice);
+    getStore().registerArray(Protected::H_ICE, &m_thick);
```

Comment:
(This is a very late comment, for context)

Is it possible to include a set of text tags in some storage structure in the class and then run over that to register all the arrays with their array references/pointers? Only reason I say that is that this pattern appears a lot in this PR and for classes where there are several of these line, for me, it starts to raise the question of how much can be automated via inherited behaviour based on class attributes/properties/variables (or something of that nature)

Reply:
That's … a very good idea. But let's put a pin in it and save it for another issue/PR.